include/Oscillator.h

```diff
@@ -165,6 +192,34 @@ class EXPORT Oscillator
 	float m_phaseOffset;
 	float m_phase;
 	const SampleBuffer * m_userWave;
+	bool m_useWaveTable;
+
+	/* Multiband WaveTable */
+	static const int WAVETABLE_LENGTH = 1024;
+	static const int MAX_FREQ = 20000; //limit to the audio spectrum
+	sample_t *m_generatedWaveTable;
+	static float *s_waveTableBandFreqs;
+	static int s_waveTablesPerWaveformCount;
+	static sample_t ***s_waveTables;
+	fftwf_plan m_fftPlan;
+	fftwf_plan m_ifftPlan;
+	fftwf_complex * m_specBuf;
+	float m_fftBuffer[WAVETABLE_LENGTH*2];
+	float m_sampleBuffer[WAVETABLE_LENGTH];
+
+	void generateSineWaveTable(int bands);
+	void generateSawWaveTable(int bands);
+	void generateTriangleWaveTable(int bands);
+	void generateSquareWaveTable(int bands);
+	void generateMoogSawWaveTable(int bands);
+	void generateExpWaveTable(int bands);
+	void generateFromFFT(int bands, float threshold);
+	int  waveTableBandFromFreq(float freq);
+	void generateWaveTables();
+	void allocWaveTables();
+	void createFFtPlans();
```

Comment:
Somewhat odd choice of cases, I guess either FFT or Fft would be better.

Reply:
revised